common/base/include/claraparabricks/genomeworks/utils/mathutils.hpp

```diff
@@ -78,6 +78,37 @@ std::int32_t int_floor_log2(T val)
     return power;
 }
 
+/// @brief Rounds up a number to the next number divisible by the given denominator. If the number is already divisible by the denominator it remains the same
+/// @param val number to round up
+/// @param roundup_denominator
+/// @tparam Integer val' type, has to be integer
+template <typename Integer>
+__host__ __device__ Integer round_up(const Integer val,
+                                     const uint32_t roundup_denominator)
```

Comment:
I use `ceiling_divide` for this functionality a lot, but I guess this primitive deserves a separate function.
Could you change the roundup_denominator type to either `Integer`, `int32_t` or `Integer2`?
Optional: maybe `roundup_next_multiple()` would be a better name?

Reply:
Renamed to `roundup_next_multiple`()`.

I changed `roundup_denominator`'s type to `int32_t` and added an assert to make sure it is positive. Function could easily be made to work with negative denominators as well, but it feels kind of weir and would probably lead to many confusions